python/ql/lib/semmle/python/ApiGraphs.qll

```diff
@@ -113,39 +104,48 @@ module API {
     }
 
     /**
-     * Gets a data-flow node corresponding to the right-hand side of a definition of the API
-     * component represented by this node.
+     * Gets a data-flow node where this value leaves the current codebase and flows into an
+     * external library (or in general, any external codebase).
```

Comment:
I'm curious about your statement in the parenthesis. What would be an example of an interaction with an external codebase, where the external codebase could not be labeled as an external library?

Reply:
External libraries are the only use-case today. But in the original use case it could also be the other way around, where the current codebase is a library and the external code-base is a user of that library (this was important when extracting summaries of a library).

There were also some other use cases that haven't been explored much, like modelling database contents with API nodes, so a project that writes to the database will generate a summary that is usable by projects that read from the database, even if the two projects are written in different languages.

Reply:
oh, that's a very good point :+1: Thanks for clarifying